org.eclipse.tips.tests/src/org/eclipse/tips/core/TipTest.java

```diff
@@ -104,8 +104,8 @@ public void testGetSubject() {
 	public void testEqualsObject() {
 		TestTip testTip = new TestTip(fProvider.getID(), HTML, SUBJECT_TIP);
 		TestTip testTipx = new TestTip(fProvider.getID(), HTML, SUBJECT_TIP);
-		assertNotNull(testTip);
-		assertEquals(testTip, testTip.equals(testTip));
+		assertFalse(testTip.equals(fTip));
+		assertTrue(testTip.equals(testTipx));
```

Comment:
better use assertEquals/NotEquals with the right arguments

Reply:
This is in a test specifically for "equals" method thus I believe it's better to call the method directly.

Reply:
Invocation of `.equals()` is made intentionally - that's what the test is exactly developed for - to test `equals()` work on these objects, as far as I understand.

The duplicating extra testing work done in the following two lines:
```
assertNotEquals(fTip, testTip);
assertEquals(testTipx, testTip);
```
but this is not the goal, it's done more to prove the `.equals()` -like assertions in two previous lines.